File: python3/1045.py
import math

a, b, c = input().split(" ")
a = float(a)
b = float(b)
c = float(c)
lista = []
lista.append(a)
lista.append(b)
lista.append(c)
lista.sort()
a = lista[2]
b = lista[1]
c = lista[0]

if(a >= b + c):
    print("NAO FORMA TRIANGULO")
else:
    if(math.pow(a, 2) == math.pow(b, 2) + math.pow(c, 2)):
        print("TRIANGULO RETANGULO")
    if(math.pow(a, 2) > math.pow(b, 2) + math.pow(c, 2)):
        print("TRIANGULO OBTUSANGULO")
    if(math.pow(a, 2) < math.pow(b, 2) + math.pow(c, 2)):
        print("TRIANGULO ACUTANGULO")
    if(a == b and b == c):
        print("TRIANGULO EQUILATERO")
    if(b == c and b != a or a == b and a != c or a == c and a != b):
        print("TRIANGULO ISOSCELES")

File: cpp/beginner/1045.cpp
#include <iostream>
#include <cmath>

using namespace std;

int main() {

    double a, b, c;
    cin >> a >> b >> c;
    double ar[] = {a, b, c};

    if (ar[0] < ar[1] ) {
        swap(ar[0], ar[1]);
    }
    if (ar[1] < ar[2]) {
        swap(ar[1], ar[2]);
    }
    if (ar[0] < ar[1]) {
        swap(ar[0], ar[1]);
    }
    a = ar[0];
    b = ar[1];
    c = ar[2];

    if (a >= (b + c)) {
        cout << "NAO FORMA TRIANGULO" << endl;
    } else {
        if (pow(a, 2) == (pow(b, 2) + pow(c, 2))) {
            cout << "TRIANGULO RETANGULO" << endl;
        }
        if (pow(a, 2) > (pow(b, 2) + pow(c, 2))) {
            cout << "TRIANGULO OBTUSANGULO" << endl;
        }
        if (pow(a, 2) < (pow(b, 2) + pow(c, 2))) {
            cout << "TRIANGULO ACUTANGULO" << endl;
        }
        if ((a == b) && (b == c)) {
            cout << "TRIANGULO EQUILATERO" << endl;
        }
        if ((a == b) && (b != c) || (a == c) && (a != b) || (b == c) && (a != b)) {
            cout << "TRIANGULO ISOSCELES" << endl;
        }
    }

    return 0;
}
